### PR TITLE
[cleanup] silence compiler warnings

### PR DESCRIPTION
**Pull request by Rechi**
## Types of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Cosmetic change (non-breaking change that doesn't touch code)
- [ ] None of the above (please explain below)

## Checklist:
- [x] My code follows the **[Code Guidelines](CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [x] All new and existing tests passed

### CODE REVIEW

File: xbmc/view/ViewStateSettings.cpp [hunk 1]
```diff
@@ -123,7 +123,7 @@ bool CViewStateSettings::Load(const TiXmlNode *settings)
   if (pElement != NULL)
   {
     int settingLevel;
-    if (XMLUtils::GetInt(pElement, XML_SETTINGLEVEL, settingLevel, (const int)SettingLevel::Basic, (const int)SettingLevel::Expert))
+    if (XMLUtils::GetInt(pElement, XML_SETTINGLEVEL, settingLevel, (int)SettingLevel::Basic, (int)SettingLevel::Expert))
```
Comment (notspiff) [inline xbmc/view/ViewStateSettings.cpp]: c++ cast?

File: xbmc/addons/PVRClient.cpp [hunk 2]
```diff
@@ -1731,9 +1731,10 @@ class CCodecIds
   CCodecIds(void)
   {
     // get ids and names
-    AVCodec* codec = NULL;
+    const AVCodec* codec = NULL;
```
Comment (notspiff) [inline xbmc/addons/PVRClient.cpp]: nullptr?

File: xbmc/view/ViewStateSettings.cpp (hunk 1 above)
Comment (Rechi) [reply xbmc/view/ViewStateSettings.cpp]: changed

File: xbmc/addons/PVRClient.cpp (hunk 2 above)
Comment (Rechi) [reply xbmc/addons/PVRClient.cpp]: changed

File: xbmc/utils/ScraperParser.cpp [hunk 3]
```diff
@@ -285,7 +285,7 @@ void CScraperParser::ParseExpression(const std::string& input, std::string& dest
 
       if (iOptional > -1) // check that required param is there
       {
-        char temp[4];
+        char temp[12];
```
Comment (fritsch) [inline xbmc/utils/ScraperParser.cpp]: This is more than a minor :-)

Comment (Rechi) [reply xbmc/utils/ScraperParser.cpp]: I know, but wasn't in the mood of creating a separate PR for the two string fixes.
All fine if I keep it in this PR?

Comment (fritsch) [reply xbmc/utils/ScraperParser.cpp]: Perfectly fine :-)

Comment (notspiff) [reply xbmc/utils/ScraperParser.cpp]: snprintf would also be enough. optional is never > 2 digits, obviously it can be evil crafted to be so but ;p

**Pull request merged**